pedl_train.py

```diff
@@ -0,0 +1,423 @@
+        notest=True,  # default False
+        rect=False,
+        resume=False,
+        single_cls=False,
+        var=None,
+        weights="weights/ultralytics68.pt",
+    )
+
+    hyp = {
+        "giou": 3.54,  # giou loss gain
+        "cls": 37.4,  # cls loss gain
+        "cls_pw": 1.0,  # cls BCELoss positive_weight
+        "obj": 49.5,  # obj loss gain (*=img_size/320 if img_size != 320)
+        "obj_pw": 1.0,  # obj BCELoss positive_weight
+        "iou_t": 0.225,  # iou training threshold
+        "lr0": 0.000579,  # initial learning rate (SGD=5E-3, Adam=5E-4) #0.00579
+        "lrf": -4.0,  # final LambdaLR learning rate = lr0 * (10 ** lrf)
+        "momentum": 0.937,  # SGD momentum
+        "weight_decay": 0.000484,  # optimizer weight decay
+        "fl_gamma": 0.5,  # focal loss gamma
+        "hsv_h": 0.0138,  # image HSV-Hue augmentation (fraction)
+        "hsv_s": 0.678,  # image HSV-Saturation augmentation (fraction)
+        "hsv_v": 0.36,  # image HSV-Value augmentation (fraction)
+        "degrees": 1.98,  # image rotation (+/- deg)
+        "translate": 0.05,  # image translation (+/- fraction)
+        "scale": 0.05,  # image scale (+/- gain)
+        "shear": 0.641,
+    }  # image shear (+/- deg)
+
+    cfg = opt.cfg
+    data = opt.data
+    img_size, img_size_test = (
+        opt.img_size if len(opt.img_size) == 2 else opt.img_size * 2
+    )  # train, test sizes
+    images = opt.images
+    epochs = opt.epochs  # 500200 batches at bs 64, 117263 images = 273 epochs
+    batch_size = opt.batch_size
+    accumulate = opt.accumulate  # effective bs = batch_size * accumulate = 16 * 4 = 64
+    weights = opt.weights  # initial training weights
+    data_dict = {
+        "classes": "80",
+        "train": "data/coco/train2017.txt",
+        "valid": "data/coco/val2017.txt",
+        "names": "data/coco.names",
+    }
+    train_path = data_dict["train"]
+    test_path = data_dict["valid"]
+
+    nc = 1 if opt.single_cls else int(data_dict["classes"])
+
+    # PEDL arguments
+    # We make some of the default hyperparameters and turn them into PEDL hyperparameters.
+
+    prebias = pedl.Constant(value=True, name="Prebias")
+    batch_size = pedl.Constant(value=16, name="batch_size")
+    init_lr = pedl.Constant(value=0.000579, name="init_lr")
+
+    # Once defined, we update the original hyperparameters with the PEDL values
+    opt.batch_size = batch_size
+    hyp["lr0"] = init_lr
+
+    config = {
+        "description": "yolov3_pytorch",
+        "searcher": {
+            "name": "single",
+            "metric": "mAP_at_0.5",
+            "max_steps": int(epochs * images * batch_size * accumulate / 100),
+            "smaller_is_better": False,
+        },
+        "max_restarts": 0,
+        "bind_mounts": [
+            {"host_path": "/home/anton/yolov3/data", "container_path": "data"}
+        ],
+        "optimizations": {"aggregation_frequency": accumulate},
+        "min_validation_period": int(
```

Comment:
I think we could put how `max_steps` and `min_validation_period` are calculated in the README.md.

Reply:
ok